src/api/api.cpp

```diff
@@ -171,6 +171,18 @@ operation create_op(const char* name, const char* attributes)
     return op;
 }
 
+std::size_t get_parameter_number(program& p) { return p.get_parameter_names().size(); }
```

Comment:
I dont see a reason to add this API as this is the same as `p.get_parameter_names().size()`, which the user can call directly.